libdnf5/transaction/transaction_sr.cpp

```diff
@@ -158,13 +160,20 @@ TransactionReplay parse_transaction_replay(const std::string & json_serialized_t
             if (json_object_object_get_ex(group, "repo_id", &value) != 0) {
                 repo_id = json_object_get_string(value);
             }
+            if (json_object_object_get_ex(group, "package_types", &value) != 0) {
+                joined_package_types = json_object_get_string(value);
+                auto package_types_vec = libdnf5::utils::string::split(joined_package_types, ",");
+                std::for_each(package_types_vec.begin(), package_types_vec.end(), libdnf5::utils::string::trim);
+                package_types = comps::package_type_from_string(package_types_vec);
+            }
```

Comment:
If the `package_types` is not in the json, it will remain uninitialized, and I think that would result in undefined behavior. So either the value should be required by the json format or it should default perhaps to the `group_package_types` configuration option.

Reply:
I have added an `else` branch to initialize it to zero/no package types.
This matches the dnf4 behavior.

I believe this is the expected behavior, when no `package_types` are specified the value is empty.
This is the same like when `"package_types":""` is in the json.

I think this should happen when installing a group with the `--no-packages` option but currently its using the default `package_types` and I consider that a bug.